src/core/client.ts

```diff
@@ -238,30 +242,36 @@ class ApiClient<R extends RawApi> {
             this.hasUsedWebhookReply = true;
             const config = createJsonPayload({ ...payload, method });
             await this.webhookReplyEnvelope.send(config.body);
-            return { ok: true, result: true };
+            return { ok: true, result: true as any };
         } else {
             const p = payload ?? {};
-            const config = formDataRequired
-                ? createFormDataPayload(p)
-                : createJsonPayload(p);
-            let res: Await<ReturnType<typeof fetch>>;
-            try {
-                res = await fetch(url, {
-                    ...this.options.baseFetchConfig,
-                    signal,
-                    ...config,
-                });
-            } catch (err) {
-                let msg = `Network request for '${method}' failed!`;
-                if (isTelegramError(err)) {
-                    msg += ` (${err.status}: ${err.statusText})`;
-                }
-                if (this.options.sensitiveLogs && err instanceof Error) {
-                    msg += ` ${err.message}`;
-                }
-                throw new HttpError(msg, err);
-            }
-            return await res.json();
+            const sensLogs = this.options.sensitiveLogs;
+
+            const abortController = new AbortController();
+            const abort = signal === undefined
```

Comment:
If combineAborts has the responsibility of checking if signal is undefined, is this check required here?

Reply:
Good point! I think it should be unified.